internal/dcache/rpc/server/server.go

```diff
@@ -36,39 +36,35 @@ package rpc_server
 import (
 	"github.com/Azure/azure-storage-fuse/v2/common"
 	"github.com/Azure/azure-storage-fuse/v2/common/log"
-	"github.com/Azure/azure-storage-fuse/v2/internal/dcache"
+	"github.com/Azure/azure-storage-fuse/v2/internal/dcache/clustermap"
+	"github.com/Azure/azure-storage-fuse/v2/internal/dcache/rpc"
 	"github.com/Azure/azure-storage-fuse/v2/internal/dcache/rpc/gen-go/dcache/service"
 	"github.com/apache/thrift/lib/go/thrift"
 )
 
-var dCacheConfig *dcache.DCacheConfig
-
 // NodeServer struct holds the Thrift server
 type NodeServer struct {
 	address string
 	server  thrift.TServer
 }
 
-// NewNodeServer creates a Thrift server for the node. It takes following parameters:
-// - address: address of the node having the format <ip>:<port>
-// - rvs: map of raw volumes of this node
-// - dCacheConfig: dCache configuration
-func NewNodeServer(address string, rvs map[string]dcache.RawVolume, config *dcache.DCacheConfig) (*NodeServer, error) {
-	// TODO: add assert for IsValidIP
-	common.Assert(address != "", "node address cannot be empty")
+// NewNodeServer creates a Thrift server for the node
+func NewNodeServer() (*NodeServer, error) {
+	nodeID, err := common.GetNodeUUID()
+	common.Assert(err == nil, "failed to get node ID: %v", err)
+
+	address := rpc.GetNodeAddressFromID(nodeID)
+	rvs := clustermap.GetMyRVs()
```

Comment:
add a TODO to call IsValidRVMap() to test rvs list and fail if it's invalid.
this can be done only after we move IsValidRVMap() to clustermap

Reply:
Added TODO